src/redux/books/books.js

```diff
@@ -27,18 +44,14 @@ const bookReducer = (state = initialState, action) => {
 export default bookReducer;
 
 // Action Creators
-let nextBookId = 2;
-export const addBook = (title, author, genre) => {
-  nextBookId += 1;
-  return (
-    {
-      type: ADD_BOOK,
-      id: toString(nextBookId),
-      title,
-      author,
-      genre,
-    });
-};
+export const addBook = (bookObj) => (
+  {
+    type: ADD_BOOK,
+    id: bookObj.id,
+    title: bookObj.title,
+    author: bookObj.author,
+    genre: bookObj.genre,
+  });
```

Comment:
- [ ] **[OPTIONAL]:** You did great working on this milestone! 💯 However, if you want to improve your code, you could follow a widely used convention for action creators: the `payload` key to send the data to the reducer, like this:

```suggestion
export const addBook = (bookObj) => (
  {
    type: ADD_BOOK,
    payload: bookObj
  });
```

This way, you don't have to list all object's properties, and in your reducer, you could implement the following change too:

```js
    case ADD_BOOK:
      return [
        ...state,
        {
          ...action.payload,
          completed: '0%',
          currentChapter: '0',
        },
      ];

```

Requiring less lines of code and making your code more readable in the long run! 🤓 🔧